### DISCUSSION TITLE
General and Multi Joomla Version Documents

### DISCUSSION THREAD

**Discussion by alann60** [max123kl/J4.docs#14] [General]
Not an issue right now but what should be done with documents that cover multiple Joomla versions, sometimes going right back to the beginning?

Should all documents be version specific?

I think many documents contain links to that type of document.

**Comment by carlitorweb** [comment]
In my opinion, we need try cover as much the current version, 4.x. And let the previous version hosted in mediawiki as a historical archive.

**Reply by max123kl** [reply]
But we should make sure that links to older documents continue to work.

**Reply by carlitorweb** [reply]
Yes, that is correct

**Reply by alann60** [reply]
So, lets say that I am creating a J4 document and want to refer to, for example another document "Article Versions" and that document contains information for J1.x, J2.x and J3.x. In this case do we add J4 and remove J1.x and J2.x or do a new document for J4.x only.

We don't want new documents with links to pages that contains information for old versions?

This is typically a problem more related to amending existing documents but not exclusively so.

My own thinking is that moving forwards, documents and linking ought to be a self contained entity for J4, J5 etc and therefore a new document should be created to link to. Whilst it will mean some duplication of a previous generic version document and more work, a self contained J4 documentation base will be starter for a J5 one, particularly if a repo of the .md versions will be available.

**Reply by max123kl** [reply]
Just a quick word about linking to other pages (I'm working on another text at the moment):
You can also link only parts of a page - there are corresponding commands in MediaWiki. I would have to search in the MW help, maybe @ceford  knows it directly.

**Reply by alann60** [reply]
That would be useful to know...

**Reply by ceford** [reply]
Several tips: 

Avoid links as much as possible. Some tutorials link to almost everything (like WikiPedia) and that can be a bad user experience. Whilst reading I reach a link and have to decide, shall I follow that link now, leave it till later, or ignore it. Several times in one paragraphs and I have forgotten the purpose of the paragraph.

If you want to link to a a heading use linkurl#headingid

In everything I have written I have specifically avoided mentioning any previous version. Joomla 3 is dead. Long live Joomla 4. So words like **This feature was introduced in Joomla 3.9...** should not be relevant to anyone reading Jooml 4 docs.
